woocommerce/woocommerce-ios
# [Woo POS] UI: Add shadows to Cart header

Closes: #13689


## Description
This PR adds a shadow element to the Cart header's bottom when: 1. There are items in cart, and 2. Merchant scrolls, both during item selection and during checkout:

https://github.com/user-attachments/assets/6320807b-55bf-43f4-a5f2-14d939e984df

## Testing
- Run POS, observe that with an empty cart of items the Cart view appears greyed-out
- Add some items to cart. Observe that still not shadow appears
- Scroll. Observe that a shadow appears below the `Cart` header. Stop scrolling to observe that the shadow disappears again upon the view returns to its original position.
- Add multiple items to cart so there's more than the current view can fit. Observe the cart still scrolls automatically to the most recent item, as well as the shadow appears in the header.

No new unit tests are needed since changes are limited to the UI.

## Changes
- Created a reusable `POSBottomShadowViewModifier` view modifier.
- Added `GeometryReader` and `PreferenceKey` usage to the VStack that contains all item rows, otherwise we cannot track scroll behaviour in order to trigger the UI effects.
- Applied the shadow effect upon two conditions: Cart must not be empty, and scroll threshold must be below the initial height by a point. I'm not quite happy about this bit and I found scroll detecting specially tricky here, so I'm open to suggestions to make this more stable.

---

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:
